build/linux/Dockerfile

```diff
@@ -1,8 +1,31 @@
+ARG BASE_IMAGE=portainer/base
 ARG GIT_COMMIT=unspecified
-FROM portainer/base:latest
+ARG GO_VERSION=1.22.7
 
+# Stage 1: Build the Go application
+FROM golang:${GO_VERSION}-alpine3.20 AS build
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the rest of the source code
+COPY --link . .
+
+# Download Go module dependencies
+RUN go mod download -x
+
+# Build the Go application
+RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} CGO_ENABLED=0 go build --installsuffix cgo --ldflags '-s' -o portainer-updater
```

Comment:
Why not just use the Makefile and `make` in the CI ? Are you experimenting to gear toward multi-stage builds across the software ? I don't recall this being something we're doing for other software.

Just a question, I'm not particularly against it, just curious. I always prefer consistency.

Reply:
Yes, it was exactly that! I was hoping to work on making multi-stage builds for everything, I know Steven also had this in mind for quite some time now. I used portainer-updater as a starting point because it is probably less complex compared to portainer or agent, wanted to start from a smaller unit and work up.

Also having everything consolidated in a multi-stage Dockerfile lets us use official docker github actions for "build-and-push" and others...which is probably a better and standardized approach. Quite keen to know any comments/suggestions etc.

I didn't remove the build-and-push script and other supporting secrets etc. from GitHub until further discussion is had about this direction.

Reply:
Sounds good to me, as long as you've checked that we are producing a similar image.

Reply:
still a work in progress, there are certain things about the image manifest I haven't been able to solve, which need needs figuring out.